documentation/source/reference.rst

```diff
@@ -0,0 +1,189 @@
+   main user-visible entry point for parsing.  *table-class*, if
+   provided, should be a subclass of :class:`<table>` to use when
+   creating a json "object".
+
+   :signature: parse-json (source, #key strict?, table-class) => (json)
+   :parameter source: An :drm:`<object>`.
+   :parameter #key strict?: An instance of :drm:`<boolean>`.
+   :parameter #key table-class: Default to :class:`<string-table>`.
+   :value json: A JSON :drm:`<object>`
+
+   :discussion:
+
+      The parse is strict by default. If ``strict?:`` :drm:`#f` is
+      used then:
+
+      - `#` is allowed as a comment character
+
+      - "\<c>" is equivalent to ``<c>``, where <c> is not a defined
+        escape character.
+
+      - trailing commas are allowed in arrays and objects
+
+.. method:: parse-json
+   :specializer: <string>
+
+   Parse a JSON object from a :drm:`<string>`.
+
+   :signature: parse-json (source, #key strict?, table-class) => (json)
+   :parameter source: An instance of :drm:`<string>`
+   :parameter #key strict?: An instance of :drm:`<boolean>`. Default to :drm:`#t`.
+   :parameter #key table-class: A subclass of :class:`<table>`.
+   :value json: An instance of :drm:`<object>`.
+
+   :example:
+
+      .. code-block:: dylan
+
+        let data = """{"a": 1, "b": 2,}""";
+        let parsed = parse-json(data);
+	let a = parsed["a"];
+
+.. method:: parse-json
+   :specializer: <stream>
+
+   Parse a JSON object from a :class:`<stream>`.
+
+   :signature: parse-json (source, #key strict?, table-class) => (json)
+   :parameter source: An instance of :class:`<stream>`.
+   :parameter #key strict?: An instance of :drm:`<boolean>`. Default to :drm:`#f`.
+   :parameter #key table-class: A subclass of :class:`<table>`.
+   :value json: An instance of :drm:`<object>`.
+
+   :example:
+
+      .. code-block:: dylan
+
+        with-open-file (fs = "data.json")
+	  let data = parse-json(fs, strict?: #f);
+	  ...
+	end;
+
+``print-json``
+--------------
+
+.. function:: print-json
+
+   Print an object in JSON format.
+
+   :signature: print-json (object, stream, #key indent, sort-keys?) => ()
+   :parameter object: The object to print. An instance of :drm:`<object>`.
+   :parameter stream: Stream on wich to do output. An instance of :class:`<stream>`.
+   :parameter #key indent: :drm:`#f` or an instance of :drm:`<integer>`.
+   :parameter #key sort-keys?: An instance of :drm:`<boolean>`.
+
+   :discussion:
+
+      If `indent` is false, *object* is printed with minimal
+      whitespace. If an integer, then use pretty printing and output
+      *indent* spaces for each indent level.
+
+      If `sort-keys:` is true, output object keys in lexicographical
+      order.
+
+``do-print-json``
+^^^^^^^^^^^^^^^^^
+
+Override this to print your own objects in JSON format. It can be
+implemented by converting objects to built-in Dylan types (tables,
+collections, etc) and calling *print* on those objects, or by
```

Comment:
"print" should be "print-json" here, I think.

Reply:
Fixed.